api/v1alpha1/hostedcluster_types.go

```diff
@@ -229,8 +229,30 @@ type HostedClusterSpec struct {
 	// provided: reconciliation is paused on the resource until the field is removed.
 	// +optional
 	PausedUntil *string `json:"pausedUntil,omitempty"`
+
+	// OLMCatalogPlacement specifies the placement of OLM catalog components. By default,
+	// this is set to management and OLM catalog components are deployed onto the management
+	// cluster. If set to guest, the OLM catalog components will be deployed onto the guest
+	// cluster.
+	//
+	// +kubebuilder:default=management
+	// +optional
+	// +immutable
+	OLMCatalogPlacement PlacementType `json:"olmCatalogPlacement,omitempty"`
 }
 
+// PlacementType is an enum specifying the placement of control plane components.
+// +kubebuilder:validation:Enum=management;guest
+type PlacementType string
+
+const (
+	// Management means components will be placed in the management cluster.
+	Management PlacementType = "management"
```

Comment:
nit: the constants should be suffixed with whatever their type is like`ManagementOLMCatalogPlacement`. They are global for the entire API package.